core/domain/classifier_domain.py

```diff
@@ -504,7 +470,126 @@ def validate(self):
                 'Expected state_name to be a string, received %s' % (
                     self.state_name))
 
-        if not isinstance(self.job_id, python_utils.BASESTRING):
+        self._algorithm_id_to_job_id_map.validate()
+
+
+class AlgorithmIDToJobIDMapping(python_utils.OBJECT):
```

Comment:
Here and elsewhere, let's do AlgorithmIdToJobIdMapping -- I think this correlates with existing conventions (e.g. StateIdMapping).

Reply:
Done